samples/nrf9160/modem_shell/src/uart/uart_shell.c

```diff
@@ -103,22 +131,9 @@ void uart_toggle_power_state(void)
 	}
 
 	if (uart0_power_state == PM_DEVICE_STATE_ACTIVE) {
-		mosh_print("Disabling UARTs");
-
-		/* allow little time for printing the notification */
-		k_sleep(K_MSEC(500));
-
-		disable_uarts();
+		k_work_submit(&uart_disable_work_q);
```

Comment:
I think there should be a print here that UARTs are going to be disabled. There are three different ways to disable/enable UARTs and all should indicate when UARTs are being disabled and enabled ("uart disable" command, modem sleep/wakeup event and button).

Reply:
There's already a print informing about toggling UART, but it's in main.c before calling `uart_toggle_power_state()`:

```
if (has_changed & button_states & DK_BTN2_MSK) {
    mosh_print("Button 2 pressed, toggling UART power state");
    uart_toggle_power_state();
}
```

https://github.com/gregersrygg/sdk-nrf/blob/65801b1df7121cce241ec1edda31f6f2175c3f79/samples/nrf9160/modem_shell/src/main.c#L111